src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -253,6 +253,11 @@ public DynamicObject multiplyTimesNegative(DynamicObject string, int times) {
             throw new RaiseException(coreExceptions().argumentError("negative argument", this));
         }
 
+        @Specialization(guards = "times < 0")
+        public DynamicObject multiplyTimesNegative(DynamicObject string, long times) {
+            throw new RaiseException(coreExceptions().argumentError("negative argument", this));
+        }
```

Comment:
You can remove the specialization above since we have an `@ImplicitCast` int => long.